config/test.exs

```diff
@@ -0,0 +1,3 @@
+import Config
+
+config :tesla, adapter: {Tesla.Adapter.Hackney, [path_encode_fun: &URI.encode/1]}
```

Comment:
For libraries, having a configuration is considered an [anti-pattern](https://hexdocs.pm/elixir/1.16.0/design-anti-patterns.html#using-application-configuration-for-libraries), so I would try to not introduce it if it wasn't already there. Speaking of which, how do the test even work? Are they supposed to be hitting real Algolia?

Reply:
So I've read that same guidance, and I've always been kind of confused by it. We configure tons of libraries using app config. Clearly there are situations where it doesn't make sense (the string splitting example given there is pretty contrived imo), but there are libraries where global configuration is useful.

But that's not really relevant for this particular config either way. I was able to confirm my hunch ([source](https://hexdocs.pm/elixir/main/Config.html)):

> Also note that the config/config.exs of a library is not evaluated when the library is used as a dependency, as configuration is always meant to configure the current project.

So the `config/config.exs` in this repo only applies when working with the library in isolation, like when running its tests. So I could probably remove this env-specific stuff and just include this Hackney config in `config/config.exs`. It won't affect consumers of the library.

And yes, the tests hit the real Algolia. They are quite slow, but I will say they gave me some confidence that this change will actually work in reality.

Reply:
I don't think this anti-pattern is a kind that alerts of something can cause actual problems for consumers. I see it as something to default to unless we have a reason to do things differently.

I believe in this particular case not having a config would highlight the fact that we cannot override tesla client/adapter at runtime (in order to use hackney in tests, but don't depend on it, or use mock adapter for some tests but not the other). You already said that we don't want to change the interface, so that's one reason to use it. Although I can see us just hardcoding `Tesla.Adapters.Mint` in the client module as another solution. I mean, if we aim for other people to use the library we'll have to support runtime configuration anyway. But if we don't, we can as well test the adapter we'll be using instead of hackney?

Reply:
I did some looking around at other libraries to make API clients using Tesla. It's kind of all over the map: some use a specific adapter and include it in their deps, some don't specify an adapter (presumably expecting you to set one in your app config).

I mostly used Hackney for the tests since it was already there, but I would be fine changing it to Mint as that's what it'll be using in our app once we switch to this version.

> I mean, if we aim for other people to use the library we'll have to support runtime configuration anyway.

Can you elaborate on what you mean by this? Given that the existing library is usable without any kind of runtime configuration, I'm not sure why it would become necessary now that we're using Tesla. It would be nice, since Tesla has some convenient ways to do that. But even if it was just shipped as-is in this PR, it would be just as usable to folks as the existing version is.